Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs

```diff
@@ -11,5 +16,38 @@ public CandidateRepository(IOptions<MongoSettings> options)
             : base(options)
         {
         }
+
+        public async Task UpsertAsync(Guid id, Guid projectId, Feedback feedback)
+        {
+            var filter = _filterBuilder.Eq(x => x.Id, id);
+            var updateBuilder = Builders<Candidate>.Update;
+
+            var updateDefinition = updateBuilder
+                 .Set("ProjectResults.$[projectResult].Feedbacks.$[feedback]", feedback);
+            var binaryProjectId = new BsonBinaryData(projectId, GuidRepresentation.Standard);
+            var binaryFeedbackUserId = new BsonBinaryData(feedback.UserId, GuidRepresentation.Standard);
+            var arrayFilters = new List<ArrayFilterDefinition>
+            {
+               new BsonDocumentArrayFilterDefinition<ProjectResult>(new BsonDocument("projectResult.ProjectId", binaryProjectId)),
+               new BsonDocumentArrayFilterDefinition<Feedback>(new BsonDocument("$and", new BsonArray
+                {
+                   new BsonDocument("feedback.UserId", binaryFeedbackUserId),
+                   new BsonDocument("feedback.Type", feedback.Type),
+                })),
+            };
+            var updateOptions = new UpdateOptions { ArrayFilters = arrayFilters };
+
+            var updateResult = await GetCollection().UpdateOneAsync(filter, updateDefinition, updateOptions);
+            if (updateResult.ModifiedCount == 0)
+            {
+                arrayFilters = new List<ArrayFilterDefinition>
+            {
```

Comment:
Format it correctly, brackets should be on level deeper

Reply:
done